### ISSUE TITLE
Operator documents issues tracking

### ISSUE THREAD

**Issue by pluskid** [apache/incubator-mxnet#3523]
This issue tries to track problems in operator documents. Please feel free to add or comments below if you find other operator documents that confuse you, lack details or does not make sense. You are also very welcome to contribute to the documentation. See also #3504 and #3513 for some guidelines of documentations.
### Activation

Ref: #3538
- [x] Softmax activation? The doc says _Softmax Activation is only available with CUDNN on GPUand will be computed at each location across channel if input is 4D._ but the `act_type` does not contain Softmax.
- [x] `softrelu`: what exactly is `softrelu`? Maybe good to add formulas for each of the activation type.
- [x] See also: add link to `LeakyReLU` for people looking for leaky activations.
### BatchNorm
- [ ] Details: provide formula and reference. See e.g. [the torch doc](https://github.com/torch/nn/blob/master/doc/simple.md#nn.BatchNormalization)
- [ ] Shared statistics: do we support BN with shared statistics for RNNs now? If yes, document it.
### BlockGrad
- [ ] Details: give an example use case in RNN unrolling.
### Concat
- [ ] `data` and `num_args`: since we specialized the calling convention for this kind of operators. It might be good to also specialize their auto-gen docstring so that they are less confusing.
### Correlation
- [ ] Details: provide formula or more explanation?
### Crop
- [ ] Formats: punctuations and spaces are missing. E.g. "... will be used This function support ...", "... we will use the h_wfor crop height ..."
- [ ] Examples: maybe need examples. I'm a bit confused by "2nd and 3rd dim", is it counting from 0 or 1?
### Custom
- [ ] Details: maybe add link to documentations on how to implement custom operators.
### Deconvolution
- [ ] Details: maybe add formula and references. I often find it is very confusing when different toolkits use different names for similar things like deconvolution, unconvolution, convolution_transpose, dialated convolution, etc.
### Dropout

Ref: #3566 
- [x] Details: I guess everybody knows dropout, but adding a formula is not bad.
### ElementWiseSum
- [ ] Doc: not sure why this operator is not taking a `data` argument. Maybe examples are needed.
- [ ] Clarification: what is the difference between this and `+`? Does this do something special?
### Embedding

Ref #3596
- [x] Details and examples: I find the doc quite confusing until I check the code in RNN examples. The inputs are not one-hot encoded, but just integer indices. The `input_dim` is the input vocabulary size, while "input dim of one-hot encoding" sounds like the axis of the multi-dim tensor that is used to do one-hot encoding.
### Flatten
- [x] Details
### IdentityAttachKLSparseReg
- [ ] Details: formula and references (if any)
### L2Normalization
- [ ] Details: "Set the l2 norm of each instance to a constant", but to what constant? 
### LRN
- [ ] Details: the doc sounds like that this is just a convolution operator.
### LeakyReLU
- [ ] Details: formula for each activation type.
### LinearRegressionOutput
- [ ] Details: formula
### LogisticRegressionOutput
- [ ] Details: formula
### MAERegressionOutput
- [ ] Details: formula
### MakeLoss
- [ ] Example
### Pooling
- [ ] Details: document expected input and output shape, add examples
### RNN
- [ ] Details: formula for each cell type?
### ROIPooling
- [ ] Details: maybe link to some references and example directory?
### Reshape
- [ ] Format: `shape` argument missing type
### SVMOutput
- [ ] Details
### Softmax
- [ ] Maybe should remove this at some point? Deprecated for a long time.
### SoftmaxOutput
- [ ] Fix: "If set to true, for a (n,k,x_1,..,x_n) dimensional input tensor, softmax will generate n_x_1_...*x_n output", does not seem to be correct, the output is still the same shape as the input, just normalized along one axis.
### SpatialTransformer
- [ ] Details: reference and example?
### SwapAxis
- [ ] Details
### UpSampling
- [ ] Related issue: https://github.com/dmlc/MXNet.jl/issues/108

**Comment by tqchen** [comment]
+1 for removing Softmax, SoftmaxActivation should be aliased as softmax

**Comment by VoVAllen** [comment]
SliceChannel and slice_axis

**Comment by VoVAllen** [comment]
Also I think there should be a clearer indexing operator

**Comment by VoVAllen** [comment]
I found the test case for sequencemask is also missing

**Comment by jiumem** [comment]
documents which confused me:
1. ElementWiseSum. why need 'num_args'?
2. UpSampling. 'sample_type' could add some explanation, why need 'num_args'? what if I give a single symbol?
3. slice_axis. could 'end' be -1? and does the result contain 'end' index?
 and seems there is no Shape, Mean operators in mxnet

**Comment by danielwaterworth** [comment]
https://github.com/apache/incubator-mxnet/issues/7811
